src/os/bluestore/NVMEDevice.cc

```diff
@@ -798,7 +799,7 @@ int NVMEDevice::open(string p)
     derr << __func__ << " unable to read " << p << ": " << cpp_strerror(r) << dendl;
     return r;
   }
-  while (r > 0 && !isalpha(buf[r-1])) {
+  while (r > 0 && !isxdigit(buf[r-1])) {
```

Comment:
Suppose that buf is "55cd2e414c93a4e2" and r is 16, the length of buf. 
In the 'while' loop, --r set r to be 15, and the number "2" is ignored,  then serial_number will be "55cd2e414c93a4e", which as the first parameter of manager.try_get().
As you know, after a long procedure, probe_cb() gets the same serial number("55cd2e414c93a4e2") via the interface of spdk_pci_device_get_serial_number(),
but now "55cd2e414c93a4e" is not equal to "55cd2e414c93a4e2", probe_cb() returns false.